.github/workflows/build-agent-full-container.yml

```diff
@@ -0,0 +1,112 @@
+name: Build Agent with Embedded Discovery Handlers
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+    - .github/actions/build-component-per-arch/**
+    - .github/actions/build-component-multi-arch/**
+    - .github/workflows/build-agent-container.yml
+    - build/containers/Dockerfile.agent
+    - agent/**
+    - shared/**
+    - version.txt
+    - build/akri-containers.mk
+    - build/akri-rust-containers.mk
+    - Makefile
+  pull_request:
+    branches: [ main ]
+    paths:
+    - .github/actions/build-component-per-arch/**
+    - .github/actions/build-component-multi-arch/**
+    - .github/workflows/build-agent-container.yml
+    - build/containers/Dockerfile.agent
+    - agent/**
+    - shared/**
+    - version.txt
+    - build/akri-containers.mk
+    - build/akri-rust-containers.mk
+    - Makefile
+  release:
+    types:
+      - published
+
+env:
+  AKRI_COMPONENT: agent-full
+  MAKEFILE_COMPONENT: agent
+
+jobs:
+
+  per-arch:
+    runs-on: ubuntu-latest
+    timeout-minutes: 40
+    strategy:
+      matrix:
```

Comment:
i wonder if agent vs agent-full could just be a dimension in the strategy matrix.